src/resource.ts

```diff
@@ -265,6 +264,18 @@ export class Resource<T = any> extends EventEmitter {
 
   }
 
+  /**
+   * Updates the state cache, and emits events.
+   *
+   * This will update the local state but *not* update the server
+   */
+  updateCache(state: State<T>) {
+
+    this.client.cache.store(state);
+    this.emit('update', state);
```

Comment:
Looks awesome! I have 2 questions:

1) What are the 'options' for the updateCache? Would one of these be for whether the user is anticipating not wanting to update the cache/do a network request? 

2) for the `emit('update')`, would this be something that is emitted on every update or can this be conditional based on client config? 

Asking mostly out of curiosity of API design and some comparing to what I remember from graphQL implementation of local cache! :P

Reply:
> What are the 'options' for the updateCache? Would one of these be for whether the user is anticipating not wanting to update the cache/do a network request?


It's a little strange (and perhaps this can use some work), but generally all these functions require a `State` object.

The `Options` variant of this is a more minimal version which makes it easy to do an update without needing the entire bulky `State`.

If you have a state, it's pretty easy to use:

    resource.put(state);

But if you don't, and want to `put` something from scratch, this is the options variant:

    resource.put({ data: { foo: bar }});

Options also allows you to specify HTTP headers and a few other things, but it doesn't let you 'skip' a network request.

> For the emit('update'), would this be something that is emitted on every update or can this be conditional based on client config?

The `update` event will be emitted in the following cases:

1. Every time you do a `put()` request
2. Every time anyone does a `refresh()`
3. Every time someone updates the local cache
4. Other, stranger conditions like resources embedded in other resources and when a feature such as `Prefer: return=representation` is used

Generally the goal is always emit this event if we positively have a new server state.

Reply:
In short, if you want to update local cache, you use `updateCache`. If you want to do a network request, you use `put()`.